packages/api-v2/src/hooks/useBalanceSubscription.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useMemo } from 'react';
+import useSubscription from '../useSubscription';
+
+type TPayload = NonNullable<Parameters<ReturnType<typeof useSubscription<'balance'>>['subscribe']>[0]>['payload'];
+
+/** A custom hook that subscribes to the balance call and continuously gets the balance for the user account(s) provided. */
+const useBalanceSubscription = () => {
+    const { data: balance_data, subscribe: _subscribe, ...rest } = useSubscription('balance');
+
+    const subscribe = useCallback(
```

Comment:
isn't it already wrapped in useCallback in use subscribe? 
if its just to change the format from payload to { payload }, then probably better to avoid such rewrite of params - while working with subsciprionts, I found that there are layers and layers of code and every single layer does something with parameters, so the less complexity we introduce to those layers, the better,

Reply:
ah I see now, its to avoid: subscribe({ payload: { loginid }})